worker/src/lib.rs

```diff
@@ -135,6 +134,93 @@ pub async fn witness_tweet(
         };
         vc.evidence = Some(OneOrMany::One(evidence));
 
+        info!("{:?}", vc);
+
+        let proof = jserr!(
+            vc.generate_proof(
+                &sk,
+                &LinkedDataProofOptions {
+                    verification_method: Some(format!("{}#controller", SPRUCE_DIDWEB)),
+                    ..Default::default()
+                }
+            )
+            .await
+        );
+        vc.proof = Some(OneOrMany::One(proof));
+        Ok(jserr!(serde_json::to_string(&vc)).into())
+    })
+}
+
+#[wasm_bindgen]
+pub async fn witness_discord(
+    secret_key_jwk: String,
+    public_key_tezos: String,
+    discord_authorization_key: String,
+    discord_handle: String,
+    channel_id: String,
+    message_id: String,
+) -> Promise {
+    use log::Level;
+    console_log::init_with_level(Level::Trace).expect("error initializing log");
+
+    future_to_promise(async move {
+        let pk: JWK = jserr!(jwk_from_tezos_key(&public_key_tezos));
+        let sk: JWK = jserr!(serde_json::from_str(&secret_key_jwk));
+        let discord_res = jserr!(
+            discord::retrieve_discord_message(discord_authorization_key, channel_id, message_id)
+                .await
+        );
+        let formatted_discord_handle = format!(
+            "{}#{}",
+            discord_res.author.username, discord_res.author.discriminator
+        );
+        let mut vc = jserr!(discord::build_discord_vc(&pk, &formatted_discord_handle));
+
+        // Check for matching handles
+        if discord_handle != formatted_discord_handle {
+            jserr!(Err(anyhow!(format!(
+                "Different Discord handle {} v. {}",
+                discord_handle, formatted_discord_handle
+            ))));
+        }
+
+        let (sig_target, sig) = jserr!(twitter::extract_signature(discord_res.content));
+        jserr!(verify_signature(&sig_target, &pk, &sig));
+
+        let mut props = HashMap::new();
+        props.insert(
+            "publicKeyJwk".to_string(),
+            jserr!(serde_json::to_value(pk.clone())),
+        );
+
+        info!("Issue Discord Credential");
+
+        let mut evidence_map = HashMap::new();
+        evidence_map.insert(
+            "handle".to_string(),
+            serde_json::Value::String(formatted_discord_handle),
+        );
+        evidence_map.insert(
+            "timestamp".to_string(),
+            serde_json::Value::String(Utc::now().to_rfc3339_opts(SecondsFormat::Millis, true)),
+        );
+
+        evidence_map.insert(
+            "channelId".to_string(),
+            serde_json::Value::String(discord_res.channel_id),
+        );
+        evidence_map.insert(
+            "messageId".to_string(),
+            serde_json::Value::String(discord_res.id),
+        );
```

Comment:
You can initialise the map from an array if you want to make it less verbose https://doc.rust-lang.org/std/collections/struct.HashMap.html#examples